File: Patrones Creacionales/Prototype/Ejemplo.js
class Prototype {
  constructor(value) {
    this.value = value;
  }

  clone() {
    return Object.assign(Object.create(Object.getPrototypeOf(this)),
      JSON.parse(JSON.stringify(this)));
  }
}

const original = new Prototype([1, 2, 3]);
const cloned = original.clone();
cloned.value.push(4);

console.log(original.value);
console.log(cloned.value);

File: Patrones Creacionales/Prototype/Ejemplo.py
import copy

class Prototype:
    def __init__(self, value):
        self.value = value

    def clone(self):
        return copy.deepcopy(self)

original = Prototype([1, 2, 3])
cloned = original.clone()
cloned.value.append(4)

print(original.value)
print(cloned.value)
